metaflow/plugins/argo/argo_workflows.py

```diff
@@ -823,10 +829,10 @@ def _compile_workflow_template(self):
                 # Set common pod metadata.
                 .pod_metadata(
                     Metadata()
+                    .labels(self._base_labels)
                     .label("app.kubernetes.io/name", "metaflow-task")
-                    .label("app.kubernetes.io/part-of", "metaflow")
                     .annotations(annotations)
-                    .labels(self.kubernetes_labels)
```

Comment:
@saikonen is there a reason that you remove these labels from the podMetadata section and instead repeat them in each task?

Reply:
the podMetadata applying to all pods from the template wasn't a good fit for setting task-pod-specific labels, e.g.
```
@kubernetes(labels={"test-label": "first-step"})
@step
def first_step(self)
   ...

@kubernetes(labels={"test-label": "second-step"})
@step
def second_step(self)
   ...
```
as on the template level we can't make a choice which deco to pull the 'common' labels from.

The previous implementation did treat the common labels from the environment variable as shared, and set them in the podMetadata. Functionally there shouldn't be any difference though whether those are set here, or individually per-pod?

Reply:
Is there an issue with repeating common labels on the pod level?

We can extend on this, as there is a todo for the future to provide separate label/annotation global env configuration for argo-workflows which would enable labeling all relevant resources (templates, sensors etc.)